lib/puppet/provider/service/systemd.rb

```diff
@@ -14,7 +14,7 @@
   confine :true => Puppet::FileSystem.exist?('/proc/1/comm') && Puppet::FileSystem.read('/proc/1/comm').include?('systemd')
 
   defaultfor :osfamily => [:archlinux]
-  defaultfor :osfamily => :redhat, :operatingsystemmajrelease => ["7", "8"]
```

Comment:
We have to keep these as strings since Facter reports strings.

Reply:
weird. didn't need to do that for the dnf provider.  I'll adjust it today

Reply:
I think array is safe, as when checking this, we do `fact_val.intern == value.to_s.downcase.intern`
https://github.com/puppetlabs/puppet/blob/main/lib/puppet/provider.rb#L284-L305